### ISSUE TITLE
Radio system splits to multiple channels on campaign

### ISSUE THREAD

**Issue by CatchFire1101** [Regalis11/Barotrauma#1659]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
In campaign mode, after multiple successful rounds, the radio system will seemingly split into multiple "channels".  Each player will only be able to hear radio voice comms and read radio chat messages from a subset of players on the server.  Proximity comms and proximity chat are unaffected.  

**Steps To Reproduce**
Unfortunately I have not been able to pinpoint the exact cause (though some combination of deaths and respawns, players leaving, new players joining server, new headset object instantiations are my guess).  

I have been regularly running a public campaign server, with multiple leaving/joining players between rounds, multiple deaths and respawns, and after at most 15 rounds into campaign (usually much sooner) the radio system will have split such that each player can only hear and speak to a subset of all players on the server over the radio system.

**Version**
V0.9.0.6
Windows

**Additional information**
Issue can generally be resolved by one of:
1) reloading the existing campaign, and at beginning of round start all comms are fixed
2) restarting server and reloading the existing campaign, at the beginning of the round all comms are fixed

**Comment by Regalis11** [comment]
Just to make sure, does everyone have batteries in their headsets? They generally run out after a couple of rounds, so that could be causing this.

**Comment by CatchFire1101** [comment]
Good thought, and yes, can confirm all players have charged batteries in their headsets.

For troubleshooting purposes, in one instance I had all players check batteries and unequip and reequip their headsets in unison, and that did not appear to resolve the issue.  Only reloading the current campaign save or restarting server and reloading the campaign save appears to fix.  

To clarify the symptoms, assume there are two players, player1 and player2 that have ended up on different "channels".  Both have batteries in their headsets and can successfully speak/radio chat to a subset of other players on the server at distance.

From player1's perspective, when player2 is nearby and speaks, he is audible and a yellow chat bubble will appear above their head, a speaker icon will appear next to his name in the player list.  When on opposite sides of the boat, player2 will not be audible and no speaker symbol will appear.  However, player2 is indeed audible for other players on the server over distance, just not player1.  In some cases Player1 can actually hear other players on the boat responding to player2 over the radio, even though they can not hear Player2 specifically.  

Sorry I can't give exact steps to replicate, however I can say it happens seemingly consistently on long campaign runs.  I have replicated it approx. 5 times.

**Comment by Regalis11** [comment]
Alright, thanks for the clarification! I'll try running some campaign rounds to hopefully reproduce this at my end.

**Comment by CatchFire1101** [comment]
I won't be running the server for a few days due to work, but if I can replicate I will try to attach logs.

Does verbose logging help troubleshooting on your end?

**Comment by Regalis11** [comment]
Unfortunately I don't think the logs would help much (even with verbose logging on), as they don't contain much extra info about the way the text/voice messages are handled. Something that could however give some extra clues are the save files, especially the files that contain the character data (including items in their inventories). You should be able to find these in "C:/Users/*user*/AppData/Local/Daedalic Entertainment GmbH/Barotrauma/".

**Comment by CatchFire1101** [comment]
[SBPublicSaveFiles.zip](https://github.com/Regalis11/Barotrauma/files/3298304/SBPublicSaveFiles.zip)

Sure thing.  At least a subset of those had the issue.

**Comment by titanrat** [comment]
@Regalis11 I'd like to ask. Maybe it's a similar problem. Yesterday (0.9.0.6 version). I tried using headphones for in-game negotiations. I've tried to take the batteries out of them or take them off. When I did that, people in the other compartments would stop hearing me. But then the traitor tried to stun me. He hit me with a shock baton and took my headphones off. When I woke up. I reported the traitor with my voice on the radio, and the entire crew even heard me on the outside of the submarine. I wasn't wearing my headphones, and the traitor took them into his inventory. Is that how the team intended to hear me through the headphones in someone else's inventory? Or is it a bug and only with the headphones equipped and charged should negotiations be possible?
